toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/MainApp.java

```diff
@@ -11,246 +11,270 @@
+		return assertProperty(AgentConfigProvider.TOAST_SCRIPTS_DIR);
+	}
+
+	private ActionListener getKillListener(){
+	    ActionListener listener = new ActionListener() {
+	        public void actionPerformed(ActionEvent event) {
+				try {
+					agentServer.unRegister();
+				} catch (UnknownHostException exception) {
+					LOG.error(exception.getMessage(), exception);
+				}finally {
+					System.exit(0);
+				}
+
 	        }
 	    };
 	    return listener;
 	}
 	
-	ActionListener getStartListener(){
+	private ActionListener getStartListener(){
 	    ActionListener listener = new ActionListener() {
 	        public void actionPerformed(ActionEvent e) {
-	        	try {
-	        		if(connectedToWebApp) {
-	        			boolean flag = false;
-		        		
-						if(verificationWebApp(chromeDriverName)) {
-							if(verificationWebApp(recorderName)) {
-								if(verificationWebApp(webAppName)) {
-									flag = true;
-									listenerStarted = true;
-									browserManager.startRecording();
-								}
-							}
-						}
-						
-						if(!flag) {
-							NotificationManager.showMessage("Unable to start recorder, please check recoder parameters !").showNotification();
-						}
-	        		}
-	        		else 
-						NotificationManager.showMessage("You have to be connected to the WebApp !").showNotification();
-				} catch (IOException e1) {
-					LOG.error(e1.getMessage(), e1);
-				}
+	        	start();
 	        }
 	    };
 	    return listener;
 	}
 	
-	ActionListener getStopListener(){
+	private void start() {
+		try {
+    		if(connectedToWebApp) {
+				if(hasValidRecordingEnvironment()) {
+					listenerStarted = true;
+					browserManager.startRecording();
+				}
+				else {
+					NotificationManager.showMessage(MainAppMessages.UNABLE_START_RECORDER).showNotification();
+				}
+    		}
+    		else {
+				NotificationManager.showMessage(MainAppMessages.UNABLE_CONNECT_WEBAPP).showNotification();
+    		}
+		} catch (IOException e1) {
+			LOG.error(e1.getMessage(), e1);
+		}
+	}
+
+	private boolean hasValidRecordingEnvironment() throws IOException {
+		return assertProperty(AgentConfigProvider.TOAST_CHROMEDRIVER_PATH) &&
+				assertProperty(AgentConfigProvider.TOAST_TEST_WEB_INIT_RECORDING_URL) &&
+				assertProperty(AgentConfigProvider.TOAST_TEST_WEB_APP_URL);
+	}
+
+	private ActionListener getStopListener(){
 	    ActionListener listener = new ActionListener() {
 	        public void actionPerformed(ActionEvent e) {
 	        	if(listenerStarted) {
 		        	browserManager.closeBrowser();
 	        	}
-	        	else
-					NotificationManager.showMessage("The recorder has not been started !").showNotification();
+	        	else{
+	        		NotificationManager.showMessage(MainAppMessages.RECORDER_NOT_STARTED).showNotification();
+	        	}
 	        }
 	    };
 	    return listener;
 	}
 	
-	ActionListener getSettingsListener(){
+	private ActionListener getSettingsListener(){
 	    ActionListener listener = new ActionListener() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1604)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "listener". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1488)